### PR TITLE
Make all implants unable to be implanted more than once

### PR DESCRIPTION
**Pull request by Simyon264**
## About the PR
This PR makes mind shields unable to be implanted more than once.
## Why / Balance

PJB complained and people waste implants by injecting already injected people.

## Technical details

Adds a new datafield to the Implanter component which is checked in implanter system.

## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

N/A

**Changelog**

:cl: Simyon
- tweak: All implants are now unable to be implanted more than once.

### CODE REVIEW

**Comment by Hanzdegloker** [comment]
Wonderful PR

**Comment by Simyon264** [comment]
Test fail is cargo spesos related, so not my fault.

File: Content.Shared/Implants/Components/ImplanterComponent.cs [hunk 1]
```diff
@@ -70,6 +70,12 @@ public sealed partial class ImplanterComponent : Component
     [DataField]
     public (string, string) ImplantData;
 
+    /// <summary>
+    /// If the <see cref="Implant"/> is already implanted, should it be allowed to be implanted again?
+    /// </summary>
+    [DataField]
+    public bool AllowMultipleImplants = true;
```
Comment (EmoGarbage404) [inline Content.Shared/Implants/Components/ImplanterComponent.cs:77]: make this false by default. i feel like this this is the exception, not the rule

File: Content.Server/Implants/ImplanterSystem.cs [hunk 2]
```diff
@@ -67,6 +79,15 @@ private void OnImplanterAfterInteract(EntityUid uid, ImplanterComponent componen
         args.Handled = true;
     }
 
+    public bool CheckSameImplant(EntityUid target, EntityUid implant)
+    {
+        if (!EntityManager.TryGetComponent<ImplantedComponent>(target, out var implanted))
```
Comment (EmoGarbage404) [inline Content.Server/Implants/ImplanterSystem.cs:84]: TryComp

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

**Comment by Simyon264** [comment]
Test fail is (again) cargo arbitrage.

**Comment by keronshb** [comment]
> Test fail is (again) cargo arbitrage.

Is this branch up to date with master?

File: Content.Shared/Implants/Components/ImplanterComponent.cs [hunk 3]
```diff
@@ -70,6 +70,12 @@ public sealed partial class ImplanterComponent : Component
     [DataField]
     public (string, string) ImplantData;
 
+    /// <summary>
+    /// If the <see cref="Implant"/> is already implanted, should it be allowed to be implanted again?
```
Comment (EmoGarbage404) [inline Content.Shared/Implants/Components/ImplanterComponent.cs:74]: This comment isnt really accurate to what it does? Should probably be more clear

File: Resources/Prototypes/Entities/Objects/Misc/implanters.yml [hunk 4]
```diff
@@ -139,6 +140,7 @@
   components:
     - type: Implanter
       implant: LightImplant
+      allowMultipleImplants: true # ok this might be a bad idea. How to win against your enemies? Crash their game with 50 lights!
```
Comment (EmoGarbage404) [inline Resources/Prototypes/Entities/Objects/Misc/implanters.yml:143]: These could all just be false, tbh. I dont really see a point in allowing multiple.

**Review by EmoGarbage404 — DISMISSED** [review]

File: Resources/Prototypes/Entities/Objects/Misc/implanters.yml (hunk 4 above)
Comment (Simyon264) [reply Resources/Prototypes/Entities/Objects/Misc/implanters.yml:143]: it's VERY funny to have 10 honk implants

File: Content.Shared/Implants/Components/ImplanterComponent.cs (hunk 3 above)
Comment (Simyon264) [reply Content.Shared/Implants/Components/ImplanterComponent.cs:74]: I don't really know how to explain it better

**Comment by Terraspark4941** [comment]
> Deny multiple implants for fun implants too.

Change PR title?

**Comment by Simyon264** [comment]
Done

**Comment by K-Dynamic** [comment]
Wouldn't this sorta prevent people from having multiple emp or multiple scram implants? They're limited-use too

**Comment by Simyon264** [comment]
Yes. This PR would prevent that.

**Comment by SlamBamActionman** [comment]
This takes away my coveted 10-bikehorn-implant-hotbar strategy. No more sounding like a Cluwne.

Probably a good thing.

**Pull request merged**